### ISSUE TITLE
[VM creation] error adding VM: error instantiating a new VM: API Error: 400: [ d1713adc-ba60-49c8-a0f0-ab740ffe635c ] Invalid network parameter: Unknown IP Addressing Mode

### ISSUE THREAD

**Issue by danielfeiler** [vmware/terraform-provider-vcd#715]
Hi there
I've run in aissue. described below.

The terraform version is 1.0.7
the vCloud Director provider version is 3.3.1

the affected resources are:
- vcd_vapp_vm

### Expected Behavior
The expected behavior will be the creation of a new VM in a existing vAPP from a template or a empty VM

### Actual Behavior
the Actual Behavior is that the creation process of the vm ends up in a error.
Error: [VM creation] error adding VM: error instantiating a new VM: API Error: 400: [ d1713adc-ba60-49c8-a0f0-ab740ffe635c ] Invalid network parameter: Unknown IP Addressing Mode
│
│   with vcd_vapp_vm.PSBAPP002,
│   on main.tf line 166, in resource "vcd_vapp_vm" "PSBAPP002":
│  166: resource "vcd_vapp_vm" "PSBAPP002" {

### Steps to Reproduce
1. create a configuration for resource vcd_vapp_vm 
2. terraform apply -auto-approve

### Important Factoids
The used Version of vCloud Director is VMware Cloud Director-Version: 10.1.2.16779297 Fri Aug 21 06:43:19 UTC 2020
If I modifie the ip allocation type for a existing an imported vm the behavior is as expected
The Error occure if I set the ip allocation mode  to DHCP, POOL, MANUAL, NONE and even if I set the Networktype to none
[terraform_IaaS01.zip](https://github.com/vmware/terraform-provider-vcd/files/7180490/terraform_IaaS01.zip)

**Comment by Didainius** [comment]
Hello,
A few questions here:
* Does the VM `PSBAPP001` work for you and it is only the `PSBAPP002` that fails?
* Could you try to shrink this example to minimum?

**Comment by danielfeiler** [comment]
> Hello,
> A few questions here:
> 
> * Does the VM `PSBAPP001` work for you and it is only the `PSBAPP002` that fails?
> * Could you try to shrink this example to minimum?

Thank you, for your answer.
The Problem is solved. The Issue results from the fact that this vApp was migrated from an old version of vCloud Director. During the migration 2 VMs were configured with the same MAC-Address.
Funnily enough, there was no error when the VM was created via the web GUI but if I've tried to rename the vApp I get exactly the described error. I found the cause as I've moved the VMs to a new vApp and then tried to deploy a new vm because I've get the output that the MAC-Address is already in use. Therefore this issue is'nt a real issue ;-)
